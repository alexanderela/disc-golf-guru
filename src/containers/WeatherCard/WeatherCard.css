.WeatherCard {
  border: 3px solid #febb00;
  color: #524949;
  width: 20rem;
  text-align: center;
  margin: 1rem auto;
  height: 24rem;
  background-color: #FFF;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  padding: 0 0 1rem 0;
  box-shadow: 0px 0px 1px 2px #Feeb00;
}

.course-weather-container {
    width: 50%;
    margin: 0 auto;
    display: flex;
    justify-content: center;
}

