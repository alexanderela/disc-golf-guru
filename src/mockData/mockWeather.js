
const mockWeather =  {
    id: 420026615,
    temp: 43.77,
    description: 'broken clouds',
    wind: 8.05,
    humidity: 76,
    icon: '01d'
}

export default mockWeather