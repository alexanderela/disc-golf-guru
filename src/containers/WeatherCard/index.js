import React, { Component } from 'react';
import './WeatherCard.css';
import Icons from '../../utilities/Icons-data';

class WeatherCard extends Component {
	constructor(props) {
		super(props);
		this.state = {

		}
	}

	render() {
		const { temp, description, wind, humidity, icon } = this.props.weather

		return(
			<div className='WeatherCard'>
				<h2 className='weather-card-header'>Current Weather</h2>	
				<img
          src={Icons[icon]} 
          alt="Icon representing current weather conditions"
          className="current-weather-icon"
        />		
        <p className='weather-info'>
					<span className='weather-info-header'>
						Temperature:
					</span> {temp}
				</p>
        <p className='weather-info'>
					<span className='weather-info-header'>
						Summary:
					</span> {description}
				</p>
        <p className='weather-info'>
					<span className='weather-info-header'>
						Wind:
					</span> {wind} mph
				</p>
        <p className='weather-info'>
					<span className='weather-info-header'>
						Humidity:
					</span> {humidity}%
				</p>
			</div>
		)
	}
}

export default WeatherCard