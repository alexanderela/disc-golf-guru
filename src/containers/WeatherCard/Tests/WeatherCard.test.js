import React from 'react';
import ReactDOM from 'react-dom';
import WeatherCard from '../';
import { shallow } from 'enzyme';
import mockWeather from '../../../mockData/mockWeather.js';

describe('WeatherCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<WeatherCard weather={mockWeather} />);
  });

  it('should render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
