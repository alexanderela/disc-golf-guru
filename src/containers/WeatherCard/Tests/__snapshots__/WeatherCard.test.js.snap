// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherCard
    weather={
      Object {
        "description": "broken clouds",
        "humidity": 76,
        "icon": "01d",
        "id": 420026615,
        "temp": 43.77,
        "wind": 8.05,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="weather-card-header"
        >
          Current Weather
        </h2>,
        <img
          alt="Icon representing current weather conditions"
          className="current-weather-icon"
          src="01d.png"
        />,
        <p
          className="weather-info"
        >
          <span
            className="weather-info-header"
          >
            Temperature:
          </span>
           
          43.77
        </p>,
        <p
          className="weather-info"
        >
          <span
            className="weather-info-header"
          >
            Summary:
          </span>
           
          broken clouds
        </p>,
        <p
          className="weather-info"
        >
          <span
            className="weather-info-header"
          >
            Wind:
          </span>
           
          8.05
           mph
        </p>,
        <p
          className="weather-info"
        >
          <span
            className="weather-info-header"
          >
            Humidity:
          </span>
           
          76
          %
        </p>,
      ],
      "className": "WeatherCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Current Weather",
          "className": "weather-card-header",
        },
        "ref": null,
        "rendered": "Current Weather",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Icon representing current weather conditions",
          "className": "current-weather-icon",
          "src": "01d.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="weather-info-header"
            >
              Temperature:
            </span>,
            " ",
            43.77,
          ],
          "className": "weather-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Temperature:",
              "className": "weather-info-header",
            },
            "ref": null,
            "rendered": "Temperature:",
            "type": "span",
          },
          " ",
          43.77,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="weather-info-header"
            >
              Summary:
            </span>,
            " ",
            "broken clouds",
          ],
          "className": "weather-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Summary:",
              "className": "weather-info-header",
            },
            "ref": null,
            "rendered": "Summary:",
            "type": "span",
          },
          " ",
          "broken clouds",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="weather-info-header"
            >
              Wind:
            </span>,
            " ",
            8.05,
            " mph",
          ],
          "className": "weather-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Wind:",
              "className": "weather-info-header",
            },
            "ref": null,
            "rendered": "Wind:",
            "type": "span",
          },
          " ",
          8.05,
          " mph",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="weather-info-header"
            >
              Humidity:
            </span>,
            " ",
            76,
            "%",
          ],
          "className": "weather-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Humidity:",
              "className": "weather-info-header",
            },
            "ref": null,
            "rendered": "Humidity:",
            "type": "span",
          },
          " ",
          76,
          "%",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="weather-card-header"
          >
            Current Weather
          </h2>,
          <img
            alt="Icon representing current weather conditions"
            className="current-weather-icon"
            src="01d.png"
          />,
          <p
            className="weather-info"
          >
            <span
              className="weather-info-header"
            >
              Temperature:
            </span>
             
            43.77
          </p>,
          <p
            className="weather-info"
          >
            <span
              className="weather-info-header"
            >
              Summary:
            </span>
             
            broken clouds
          </p>,
          <p
            className="weather-info"
          >
            <span
              className="weather-info-header"
            >
              Wind:
            </span>
             
            8.05
             mph
          </p>,
          <p
            className="weather-info"
          >
            <span
              className="weather-info-header"
            >
              Humidity:
            </span>
             
            76
            %
          </p>,
        ],
        "className": "WeatherCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Current Weather",
            "className": "weather-card-header",
          },
          "ref": null,
          "rendered": "Current Weather",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Icon representing current weather conditions",
            "className": "current-weather-icon",
            "src": "01d.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="weather-info-header"
              >
                Temperature:
              </span>,
              " ",
              43.77,
            ],
            "className": "weather-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Temperature:",
                "className": "weather-info-header",
              },
              "ref": null,
              "rendered": "Temperature:",
              "type": "span",
            },
            " ",
            43.77,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="weather-info-header"
              >
                Summary:
              </span>,
              " ",
              "broken clouds",
            ],
            "className": "weather-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Summary:",
                "className": "weather-info-header",
              },
              "ref": null,
              "rendered": "Summary:",
              "type": "span",
            },
            " ",
            "broken clouds",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="weather-info-header"
              >
                Wind:
              </span>,
              " ",
              8.05,
              " mph",
            ],
            "className": "weather-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Wind:",
                "className": "weather-info-header",
              },
              "ref": null,
              "rendered": "Wind:",
              "type": "span",
            },
            " ",
            8.05,
            " mph",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="weather-info-header"
              >
                Humidity:
              </span>,
              " ",
              76,
              "%",
            ],
            "className": "weather-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Humidity:",
                "className": "weather-info-header",
              },
              "ref": null,
              "rendered": "Humidity:",
              "type": "span",
            },
            " ",
            76,
            "%",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
